pyincore/analyses/mmsacge/mmsacge.py

```diff
@@ -0,0 +1,169 @@
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            Tuple containing two numpy arrays. The first array is the base capital for different factors and the second is the base capital
+        """
+        logger.info(f"Parsing {file_name}")
+        base_caps = pd.read_excel(file_name, sheet_name=[1, 2, 3, 4, 5])
+        base_cap_factors: List[np.ndarray] = []
+        base_cap: np.ndarray = np.array(base_caps[5]["baseKAP"]).reshape(
+            1, -1
+        )  # 1 x K array K = number of sectors in the model
+
+        for i in range(1, 5):
+            base_cap_factors.append(
+                np.array(base_caps[i][0]).reshape(-1, 1)
+            )  # k_i x 1 array k_i = number of sectors k for a factor i
+
+        return base_cap_factors, base_cap
+
+    def parse_coeff(
+        self,
+        filenames: Dict[str, str]
+    ) -> Tuple[Dict[str, np.ndarray], Dict[str, List[str]]]:
+        """parse_coeff Function to parse the model coefficients.
+
+        Parameters
+        ----------
+        filenames : Dict[str, str]
+            Dictionary containing the factor name and the path to the coefficient file
+
+        Returns
+        -------
+        Tuple[Dict[str, np.ndarray], Dict[str, List[str]]]
+            Tuple containing two dictionaries. The first dictionary contains the model coefficients for each factor and the second dictionary contains the sectors for each factor
+        """
+        model_coeffs: Dict[str, np.ndarray] = {}
+        sectors: Dict[str, List[str]] = {}
+        for factor, filename in filenames.items():
+            logger.info(f"Parsing {filename}")
+            model_coeff_df: pd.DataFrame = pd.read_csv(filename)
+            sectors[factor] = list(model_coeff_df.columns[5:])
+            model_coeffs[factor] = model_coeff_df[
+                model_coeff_df.columns[4:]
+            ].to_numpy()[
+                1:, :
+            ]  # skip the first row as it contains the total value model and its not needed in output
+
+        return model_coeffs, sectors
+
+    def parse_files(
+        self,
+    ) -> Tuple[Dict[str, List[str]], np.ndarray, np.ndarray, Dict[str, np.ndarray]]:
+        """parse_files Utility function to parse the input files
+
+        Returns
+        -------
+        Tuple[Dict[str, List[str]], np.ndarray, np.ndarray, Dict[str, np.ndarray]]
+            Returns a tuple containing the following:
+            1. sectors: Dictionary containing the sectors for each factor
+            2. base_cap_factors: List of numpy arrays containing the base capital for each factor
+            3. base_cap: Numpy array containing the base capital
+            4. model_coeffs: Dictionary containing the model coefficients for each factor
+        """
+        logger.info("Parsing input files...")
+        base_cap_factors, base_cap = self.parse_base_vals(os.path.join(pyglobals.PYINCORE_PACKAGE_HOME, "analyses", "mmsacge", "base_val_shelby_MMSA.xlsx"))
```

Comment:
Does it have to be xlsx? I can see there are many sheets inside.

Reply:
yeah, so I had a chat with Jong and he said we will make others to give us CSV :P let them do the work hahaha.... I will simplify the code and push it this week.